pymodm/base/options.py

```diff
@@ -86,8 +86,14 @@ def get_field_from_attname(self, attname):
 
     def add_field(self, field_inst):
         """Add or replace a given Field."""
-        orig_field = (self.get_field(field_inst.mongo_name) or
-                      self.get_field_from_attname(field_inst.attname))
+        try:
+            orig_field = self.get_field(field_inst.mongo_name)
+        except Exception:
+            # FieldDoesNotExist, etc. may be raised by subclasses.
+            try:
+                orig_field = self.get_field_from_attname(field_inst.attname)
```

Comment:
Actually, I'm not sure why this check is necessary at all. Why do we need to do

    orig_field = self.get_field_from_attname(field_inst.attname)

?

There's no way that we're going to raise the `InvalidModel` exception below, since necessarily `field_inst.attname == orig_field.attname`. Am I missing something?

The [test](https://github.com/mongodb/pymodm/blob/5b20864de10e6d41ddd54b762de9c7adf3d38a00/test/test_model_basic.py#L62-L74) added by this code originally continues to pass in the absence of retrieving `orig_field` by attname.